### PR TITLE
refactor: unused references cleanup

### PR DESCRIPTION
**Pull request by JoaoRodriguesGithub**
# Description

This refactor aims to remove unused references.

## How Has This Been Tested?

Builds with success

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [ ] I have added tests to cover my changes
-   [ ] I have made corresponding changes to the documentation

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by ruiqbarbosa — APPROVED** [review]

File: samples/KafkaFlow.Sample.ConsumerThrottling/KafkaFlow.Sample.ConsumerThrottling.csproj [hunk 1]
```diff
@@ -17,7 +17,6 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <ProjectReference Include="..\..\src\KafkaFlow.Admin\KafkaFlow.Admin.csproj" />
```
Comment (brmagadutra) [inline samples/KafkaFlow.Sample.ConsumerThrottling/KafkaFlow.Sample.ConsumerThrottling.csproj:20]: Wouldn't the removal of Admin packages prevent access to the dashboard? Couldn't be a case of unreferenced at the build time but used in execution?

File: src/KafkaFlow.Admin.Dashboard/KafkaFlow.Admin.Dashboard.csproj [hunk 2]
```diff
@@ -23,7 +23,6 @@
   </ItemGroup>
   
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.FileProviders.Embedded" Version="6.0.20" />
```
Comment (brmagadutra) [inline src/KafkaFlow.Admin.Dashboard/KafkaFlow.Admin.Dashboard.csproj:26]: I think @filipeesch can shed some light here, if this is necessary or not.

File: src/KafkaFlow.SchemaRegistry/KafkaFlow.SchemaRegistry.csproj [hunk 3]
```diff
@@ -9,7 +9,6 @@
 
     <ItemGroup>
         <PackageReference Include="Confluent.SchemaRegistry" Version="2.1.1" />
-        <PackageReference Include="Microsoft.IO.RecyclableMemoryStream" Version="2.3.2" />
```
Comment (brmagadutra) [inline src/KafkaFlow.SchemaRegistry/KafkaFlow.SchemaRegistry.csproj:12]: I think @filipeesch can shed some light here, if this is necessary or not.

File: src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro.csproj [hunk 4]
```diff
@@ -14,8 +14,6 @@
 
     <ItemGroup>
         <PackageReference Include="Apache.Avro" Version="1.11.0" />
-        <PackageReference Include="Confluent.Kafka" Version="2.1.1" />
```
Comment (brmagadutra) [inline src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro.csproj:17]: Don't we need this @filipeesch?

File: src/KafkaFlow/KafkaFlow.csproj [hunk 5]
```diff
@@ -10,7 +10,6 @@
         <PackageReference Include="Confluent.Kafka" Version="2.1.1" />
         <PackageReference Include="Microsoft.IO.RecyclableMemoryStream" Version="2.3.2" />
         <PackageReference Include="System.Threading.Channels" Version="6.0.0" />
-        <PackageReference Include="System.Memory" Version="4.5.5" />
```
Comment (brmagadutra) [inline src/KafkaFlow/KafkaFlow.csproj:13]: I think @filipeesch can shed some light here, if this is necessary or not.

**Review by brmagadutra — APPROVED** [review]

**Review by erik-catalao — APPROVED** [review]
LGTM

**Review by gsferreira — APPROVED** [review]

File: src/KafkaFlow/KafkaFlow.csproj (hunk 5 above)
Comment (filipeesch) [reply src/KafkaFlow/KafkaFlow.csproj:13]: If the code is still compiling successfully and the samples and tests are working, we can safely remove it.

File: src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro.csproj (hunk 4 above)
Comment (filipeesch) [reply src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro.csproj:17]: https://github.com/Farfetch/kafkaflow/pull/486#discussion_r1422258927

File: src/KafkaFlow.SchemaRegistry/KafkaFlow.SchemaRegistry.csproj (hunk 3 above)
Comment (filipeesch) [reply src/KafkaFlow.SchemaRegistry/KafkaFlow.SchemaRegistry.csproj:12]: https://github.com/Farfetch/kafkaflow/pull/486#discussion_r1422258927

File: src/KafkaFlow.Admin.Dashboard/KafkaFlow.Admin.Dashboard.csproj (hunk 2 above)
Comment (filipeesch) [reply src/KafkaFlow.Admin.Dashboard/KafkaFlow.Admin.Dashboard.csproj:26]: https://github.com/Farfetch/kafkaflow/pull/486#discussion_r1422258927

File: samples/KafkaFlow.Sample.ConsumerThrottling/KafkaFlow.Sample.ConsumerThrottling.csproj (hunk 1 above)
Comment (filipeesch) [reply samples/KafkaFlow.Sample.ConsumerThrottling/KafkaFlow.Sample.ConsumerThrottling.csproj:20]: https://github.com/Farfetch/kafkaflow/pull/486#discussion_r1422258927

**Review by brmagadutra — APPROVED** [review]

**Review by lpsc82 — APPROVED** [review]
👍

**Pull request merged**